sdks/java/core/src/test/java/org/apache/beam/sdk/testing/TestStreamTest.java

```diff
@@ -105,6 +105,9 @@ public void testLateDataAccumulating() {
             .advanceWatermarkTo(instant.plus(Duration.standardMinutes(6)))
             // These elements are late but within the allowed lateness
             .addElements(TimestampedValue.of(4L, instant), TimestampedValue.of(5L, instant))
+            .advanceWatermarkTo(instant.plus(Duration.standardMinutes(10)))
```

Comment:
@kennknowles Maybe you could answer this? I'm wondering if this is an issue of the Spark streaming runner (and how this is handled by other runners) or if it's a lack of my own understanding.

Reply:
Just responding to let you know I have been on vacation and I will look at this later today.

Reply:
Thanks a lot! I'm off as well for a bit, so no rush on this.

Reply:
@kennknowles Finally back to this, if you could have a look it would be great:)